packages/cli/src/commands/oracle/report.ts

```diff
@@ -0,0 +1,64 @@
+import { CeloContract } from '@celo/contractkit'
+import { flags } from '@oclif/command'
+import BigNumber from 'bignumber.js'
+import { BaseCommand } from '../../base'
+import { displaySendTx } from '../../utils/cli'
+import { Flags } from '../../utils/command'
+
+export default class ReportPrice extends BaseCommand {
+  static description =
+    'Report the price of Celo Gold in a specified token (currently just Celo Dollar, aka: "StableToken")'
+
+  static args = [
+    {
+      name: 'token',
+      required: true,
+      description: 'Token to report on',
+      options: [CeloContract.StableToken],
+    },
+  ]
+  static flags = {
+    ...BaseCommand.flags,
+    from: Flags.address({ required: true, description: 'Address of the oracle account' }),
+    numerator: flags.string({
+      required: true,
+      description: 'Amount of the specified token equal to the amount of cGLD in the denominator',
+    }),
+    denominator: flags.string({
+      required: false,
+      description: 'Amount of cGLD equal to the numerator. Defaults to 1 if left blank',
+    }),
+  }
+
+  static example = [
+    'report --token StableToken --numerator 1.02 --from 0x8c349AAc7065a35B7166f2659d6C35D75A3893C1',
+    'report --token StableToken --numerator 102 --denominator 100 --from 0x8c349AAc7065a35B7166f2659d6C35D75A3893C1',
+  ]
+
+  async run() {
+    const res = this.parse(ReportPrice)
+    const sortedOracles = await this.kit.contracts.getSortedOracles()
+    let numerator = new BigNumber(res.flags.numerator)
+    let denominator = new BigNumber(res.flags.denominator || 1)
+    if (numerator.decimalPlaces() > 0) {
+      const multiplier = new BigNumber(10).pow(numerator.decimalPlaces()).toNumber()
+      numerator = numerator.multipliedBy(multiplier)
+      denominator = denominator.multipliedBy(multiplier)
+    }
+
+    await displaySendTx(
+      'sortedOracles.report',
+      await sortedOracles.report(
+        res.args.token,
+        numerator.toNumber(),
```

Comment:
this should be `toFixed()`. same on the other line... we could lose precision with toNumber

Reply:
I can't use `toFixed()` here without also changing `contractkit`. The function in the wrapper expects the type of `numerator` and `denominator` to be numbers, and specifically, integers. Lines 46-49 do some processing of the initial values to convert a float value into a relationship between integers.

For example, if the command line args are: `--numerator 1.081 --denominator 1`, they'll become `numerator = 1081`, `denominator = 1000`.

Given this, do you still think there's risk of loss of precision?

Reply:
oh, nevermind then. maybe put a TODO comment